plugin/backend_linux.go

```diff
@@ -200,15 +200,29 @@ func (pm *Manager) Set(name string, args []string) error {
 // CreateFromContext creates a plugin from the given pluginDir which contains
 // both the rootfs and the config.json and a repoName with optional tag.
 func (pm *Manager) CreateFromContext(ctx context.Context, tarCtx io.Reader, options *types.PluginCreateOptions) error {
+	repoName := options.RepoName
+	ref, err := distribution.GetRef(repoName)
+	if err != nil {
+		return err
+	}
+
+	name := ref.Name()
+	tag := distribution.GetTag(ref)
 	pluginID := stringid.GenerateNonCryptoID()
 
+	p := v2.NewPlugin(name, pluginID, pm.runRoot, pm.libRoot, tag)
+
+	if v, _ := pm.pluginStore.GetByName(p.Name()); v != nil {
```

Comment:
Thanks @anusha-ragunathan for the review. In `Add()`, the name is obtained through `p.Name()`, which is actually `name:tag`.

If we only pass `name` to `GetByName(name)` here, then the collision will not be detected.

So I used `p.Name()` to have the same `name:tag` both in `Add()` and here `GetByName(p.Name())`.